iam_tags.go

```diff
@@ -3,11 +3,18 @@ package main
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/Cox-Automotive/alks-go"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
+type TagMap map[string]interface{}
```

Comment:
Let me check on this.  Thats how aws always wrote theirs, so I mimicked that, but it did bother me the entire time.  I'll go back and figure out if thats actually necessary.

Reply:
Ok so I tried to go and change this around.  It did not work because the schema.ResourceDiff functions I'm using in SetTagsDiff error out if I try to make it a map[string][string]

'panic: interface conversion: interface {} is map[string]interface {}, not map[string]string'

I could implement some type switching loop that loops through the map, writes to a new map, converting the interface{} value to string, but I'd have to switch back and forth multiple times.  That seems messier then just using map[string]interface{}

Let me know if you disagree or want to test anything, I just spent about 45 mins trying to get it to work with [string]string